openfl/federated/task/runner_keras.py

```diff
@@ -348,46 +361,23 @@ def set_tensor_dict(self, tensor_dict, with_opt_vars):
         if with_opt_vars is False:
             # It is possible to pass in opt variables from the input tensor
             # dict. This will make sure that the correct layers are updated
-            model_weight_names = [weight.name for weight in self.model.weights]
+            model_weight_names = self._get_weights_names(self.model)
             model_weights_dict = {name: tensor_dict[name] for name in model_weight_names}
             self._set_weights_dict(self.model, model_weights_dict)
         else:
-            model_weight_names = [weight.name for weight in self.model.weights]
+            model_weight_names = self._get_weights_names(self.model)
             model_weights_dict = {name: tensor_dict[name] for name in model_weight_names}
-            opt_weight_names = [weight.name for weight in self.model.optimizer.weights]
+            opt_weight_names = self._get_weights_names(self.model.optimizer)
             opt_weights_dict = {name: tensor_dict[name] for name in opt_weight_names}
             self._set_weights_dict(self.model, model_weights_dict)
             self._set_weights_dict(self.model.optimizer, opt_weights_dict)
 
     def reset_opt_vars(self):
         """Resets the optimizer variables."""
-        for var in self.model.optimizer.variables():
+        for var in self.model.optimizer.variables:
             var.assign(tf.zeros_like(var))
```

Comment:
Re: learning rate resetting to 0
This line seems to be the cause of that issue. It's a legacy line so I am not sure the context behind it, but for some reason this line will reset every variable of the optimizer (including learning rate) to 0, which is why the model is learning when `self.opt_treatment == "RESET"`. The torch task runner actually seems to just pass through this method entirely ([see](https://github.com/securefederatedai/openfl/blob/develop/openfl/federated/task/runner_pt.py#L453))

FYI, we should also get rid of any `tf` calls or imports to remove any dependency on tensorflow in the keras task runner

Reply:
this code is removed.